src/map/tiles.c

```diff
@@ -20,6 +20,7 @@
 #include "map/random.h"
 #include "map/terrain.h"
 #include "scenario/map.h"
+#include "widget/city_draw_highway.h"
```

Comment:
Calling drawing code from here doesn't make much sense from a code flow standpoint. This needs some refactoring. I think that in last resort it would be preferable to have the image ids stored in this file and have the draw code request the base image from here. Either that or we:

- store the base image id on both places;
- don't store it at all, the overhead is not that significant

Reply:
I think this is your main complaint for the PR and I totally was thinking “man @crudelios is not gonna like this” as I was writing it. How do you feel about maybe putting the highway image ids into image.c or assets.c? It’s important to me that there be one place where the image ids are stored. We can do it in this file, but it doesn’t feel like the right place for it either 🤔

Reply:
Having a general image id cache from assets based on enums isn't such a bad idea.

The enum would be the array index of a structure that had the asset group name, asset id name and actual image index.

We could then have an `asset_get_image_from_id_cache(<the enum I mentioned above)`, which would retrieve the image id of the structure of the array index. If the image id was still `0`, then it would find the proper id from `asset_get_image(entry.group, entry.name)`.

Reply:
This sounds pretty good to me. The only thing I'd change is I would not do lazy initialization, as I feel the same way about that as you do about wasting bytes. Seems like there must be some sort of ID check in the code at startup, right? How does the game know when assets are missing?

Reply:
There is no ID check at startup, only when needed.

To figure if there are missing assets, the game just looks for the roadblock asset. If it's missing, it'll complain all assets are missing.

Apart from that, since we'd manually populate that struct it's easy to get the proper image ids on boot. Just traverse the asset id cache array and fill the ids after loading all the assets, on `assets_init`.